src/HID-APIs/KeyboardAPI.hpp

```diff
@@ -154,14 +154,39 @@ size_t KeyboardAPI::set(uint8_t k, bool s){
 	}
 
 	// Read key from ascii lookup table
-	k = pgm_read_byte(_asciimap + k);
-	auto ret = set(KeyboardKeycode(k & ~SHIFT), s);
+	uint16_t key = pgm_read_word(_asciimap + k);
```

Comment:
This of course adds some more boilerplate for non us layouts. That is what I was trying to avoid. Just saying. Maybe using another wrapper could help still having a compact us layout and larger german layout at the same time.

Reply:
I don't think so. Reading 2 instead of 1 Byte once every Keystroke won't slow down anything, at least compared to the rest of the code. A few more comparisons for ALT, CTRL etc. are also neglible on those arduinos.